### DISCUSSION TITLE
Do I have to use drift instead of moor_flutter?

### DISCUSSION THREAD

**Discussion by RakaAlrian** [simolus3/drift#1624] [Q&A] [answered]
I have an app already available on playstore using moor_flutter, should I change/migrate it to drift? Is moor_flutter already deprecated, cause I found that moor_generator isn't compatible with analyer 3.0.0, thanks.

**Comment by kuhnroyal** [comment]
Drift is just a rename, not a new library. If you are using the latest moor, then you already have drift as dependency. Check you `pubspec.lock`. The migration is straight forward, mostly some imports and regeneration of sources.

**Comment by simolus3** [comment, marked as answer]
Adding to what @kuhnroyal has mentioned: There's a tool to [automatically migrate](https://drift.simonbinder.eu/name/#automatic-migration) your project from `moor` to `drift`. It has been working pretty well for my projects and should help you migrate to `drift` in minutes if you're interested. Drift databases are also fully compatible to moor databases, you don't have to worry about breaking existing users.

Using the moor packages is fine for now, but I recommend migrating to drift if you can. There probably won't be a moor 5.x, I'll switch everything to drift in the next breaking release. So migrating isn't urgent, but generally recommended.

Regarding your concern about moor_generator not supporting the latest analyzer: I have just published `drift_dev` version 1.2.1 fixing this problem. As moor_generator depends on `drift_dev`, you can just run `pub upgrade` and it should hopefully resolve to the latest analyzer.

**Reply by RakaAlrian** [reply]
Thank you for your response
I have some more questions if you don't mind, why this -`FlutterQueryExecutor.inDatabaseFolder(path: "app_db.sqlite",logStatements: true)`- method no longer exist in drift? And why drift require sqlite3_flutter_libs to work on some devices but moor_flutter doesn't?

**Reply by simolus3** [reply]
You don't have to use `sqlite3_flutter_libs` and a `NativeDatabase` for drift - you can keep using a `FlutterQueryExecutor` from `moor_flutter` even if the rest of your app uses drift. Using a `NativeDatabase` is the recommended approach for new users, but if the old implementation has been serving you well I don't see a strong reason for switching.

At the moment the situation is a bit confusing because `moor_flutter` still has moor in its name and there's no `drift_flutter` yet. But I'll keep supporting `moor_flutter` until I have a drift-named package built with `sqflite`.

**Reply by RakaAlrian** [reply]
Very well explanation, thank you very much!

**Reply by simolus3** [reply]
The `moor_flutter` package has now been replaced with the new `drift_sqflite` package (which I hope is also a much better name for what it's doing: The core `drift` package works on Flutter just fine). If you're using `moor_flutter` with drift, upgrade to `drift_dev` version `1.5.1` and run `drift_dev migrate` again to replace usages of `moor_flutter` and `FlutterQueryExecutor`.

**Reply by RakaAlrian** [reply]
Try to migrate from `moor_flutter` to `drift_sqflite` using this command `flutter pub run moor_generator migrate`. Migration success, `drift_sqflite: ^1.0.0` and `drift_dev: ^1.0.0` automatically added to my pubspec.yaml, and then I run `flutter pub get`. But this automatically added import `package:drift_sqflite/drift_sqflite.dart` is error. 😕

**Reply by simolus3** [reply]
What error are you getting with the import? The file should exist.

**Reply by RakaAlrian** [reply]
I just ran the same commands today and it has no more error, `drift_sqflite.dart` file is detected normally. Maybe there was cache problem on my vscode yesterday. Sorry to bother you. 😁
